tiledb/api/c_api/array_schema_evolution/array_schema_evolution_api_experimental.h

```diff
@@ -0,0 +1,50 @@
+/**
+ * @file tiledb/api/c_api/array_schema/array_schema_evolution_api_experimental.h
```

Comment:
It would be better if the array-evolve C API function were also declared here, and that there be a `.cc` for its implementation. I don't care one way or another if it happens in this PR.